### PR TITLE
fix: Minor UI & Feature Set Changes

### PR DESCRIPTION
**Pull request by HamzaIsrar12**
The following views or features are hidden as part of this PR:

- Device-Specific Arithmetic Exception on Pixel
- Fixes the View All Course Count
- Checkbox State on SignUp Screen
- Remove Calendar Sync from Settings Screen
- Remove Search Icon from All Courses (Grid View)
- Disabled the New Notification API
- Fixed Notification Permission Dialog on Course Home
- Text Field Keyboard Padding in Discovery Webview
- Contact Support Email Subject on Settings Screen
- Center Align Edit Profile Fields
- Fixed the User Agent Issue on Component Webviews

### CODE REVIEW

File: course/src/main/java/org/openedx/course/presentation/unit/html/HtmlUnitViewModel.kt [hunk 1]
```diff
@@ -27,6 +32,13 @@ class HtmlUnitViewModel(
     val isCourseUnitProgressEnabled get() = config.getCourseUIConfig().isCourseUnitProgressEnabled
     val apiHostURL get() = config.getApiHostURL()
     val cookieManager get() = edxCookieManager
+    val appUserAgent: String
+        get() {
+            val appName = resourceManager.getString(R.string.app_name)
```
Comment (farhan-arshad-dev) [inline course/src/main/java/org/openedx/course/presentation/unit/html/HtmlUnitViewModel.kt:37]: better to move the app name within the`AppData`.

File: course/src/main/java/org/openedx/course/presentation/unit/html/HtmlUnitViewModel.kt [hunk 2]
```diff
@@ -27,6 +32,13 @@ class HtmlUnitViewModel(
...
+            val applicationId = appData.applicationId
+            val versionName = appData.versionName
+            return "http.agent $appName/$applicationId/$versionName"
```
Comment (farhan-arshad-dev) [inline course/src/main/java/org/openedx/course/presentation/unit/html/HtmlUnitViewModel.kt:40]: need to do the same for other webviews

Comment (farhan-arshad-dev) [inline course/src/main/java/org/openedx/course/presentation/unit/html/HtmlUnitViewModel.kt:40]: `http.agent` only used for back-end APIs. not for the webview. please confirm on your end too.

**Review by farhan-arshad-dev — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/openedx/core/domain/model/Progress.kt [hunk 3]
```diff
@@ -11,11 +11,12 @@ data class Progress(
 ) : Parcelable {
 
     @IgnoredOnParcel
-    val value: Float = try {
-        assignmentsCompleted.toFloat() / totalAssignmentsCount.toFloat()
-    } catch (_: ArithmeticException) {
-        0f
-    }
+    val value: Float
```
[the remark this answers is not in the document]
Comment (HamzaIsrar12) [reply core/src/main/java/org/openedx/core/domain/model/Progress.kt:14]: I would have opted for try-catch if the value of `totalAssignmentsCount` was in float or double.

File: auth/src/main/java/org/openedx/auth/data/model/RegistrationFields.kt [hunk 4]
```diff
@@ -24,6 +24,8 @@ data class RegistrationFields(
         val exposed: Boolean?,
         @SerializedName("required")
         val required: Boolean?,
+        @SerializedName("defaultValue")
+        val defaultValue: Boolean?,
```
[the remark this answers is not in the document]
Comment (HamzaIsrar12) [reply auth/src/main/java/org/openedx/auth/data/model/RegistrationFields.kt:28]: I agree, though to our luck the field wasn't used in other views and it also won't cause issue during parsing.

File: core/src/main/java/org/openedx/core/domain/model/RegistrationField.kt [hunk 5]
```diff
@@ -15,7 +15,7 @@ data class RegistrationField(
     val restrictions: Restrictions,
     val options: List<Option>,
     val errorInstructions: String = "",
-    val defaultValue: Boolean = false,
+    val defaultValue: Boolean,
```
[the remark this answers is not in the document]
Comment (HamzaIsrar12) [reply core/src/main/java/org/openedx/core/domain/model/RegistrationField.kt:18]: I have set the default value in the data model during mapping.

File: dashboard/src/main/java/org/openedx/courses/presentation/AllEnrolledCoursesView.kt [hunk 6]
```diff
@@ -279,10 +275,7 @@ private fun AllEnrolledCoursesView(
                                             layoutDirection
                                         ),
                                         end = contentPaddings.calculateEndPadding(layoutDirection)
-                                    ),
-                                onSearchClick = {
-                                    onAction(AllEnrolledCoursesAction.Search)
-                                }
+                                    )
                             )
```
[the remark this answers is not in the document]
Comment (HamzaIsrar12) [reply dashboard/src/main/java/org/openedx/courses/presentation/AllEnrolledCoursesView.kt:279]: The feature was intended to search from enrolled courses. I will remove the related files as well.

File: course/src/main/java/org/openedx/course/presentation/unit/html/HtmlUnitViewModel.kt (hunk 2 above)
Comment (HamzaIsrar12) [reply course/src/main/java/org/openedx/course/presentation/unit/html/HtmlUnitViewModel.kt:40]: I haven't seen it on other webviews. Maybe we can skip this for now.

File: dashboard/src/main/java/org/openedx/courses/presentation/AllEnrolledCoursesViewModel.kt [hunk 7]
```diff
@@ -158,12 +158,6 @@ class AllEnrolledCoursesViewModel(
         }
     }
 
-    fun navigateToCourseSearch(fragmentManager: FragmentManager) {
-        dashboardRouter.navigateToCourseSearch(
```
Comment (farhan-arshad-dev) [inline dashboard/src/main/java/org/openedx/courses/presentation/AllEnrolledCoursesViewModel.kt:162]: no use of `navigateToCourseSearch` in `dashboardRouter` better to remove this.

**Review by farhan-arshad-dev — CHANGES_REQUESTED** [review]

File: auth/src/main/java/org/openedx/auth/presentation/ui/CheckboxField.kt [hunk 8]
```diff
@@ -14,18 +14,24 @@ import androidx.compose.runtime.setValue
...
 internal fun CheckboxField(
-    text: String,
-    defaultValue: Boolean,
+    field: RegistrationField,
     onValueChanged: (Boolean) -> Unit
 ) {
-    var checkedState by remember { mutableStateOf(defaultValue) }
+    var checkedState by remember {
+        mutableStateOf(
+            field.placeholder.takeIf { it.isNotEmpty() }?.toBoolean() ?: field.defaultValue
```
[the remark this answers is not in the document]
Comment (HamzaIsrar12) [reply auth/src/main/java/org/openedx/auth/presentation/ui/CheckboxField.kt:33]: If the user hasn't interacted with the checkbox, the placeholder will be empty, so we use the default value. If the user has enabled or disabled the checkbox, we prioritize the value stored in the placeholder.

**Review by farhan-arshad-dev — APPROVED** [review]

**Review by k1rill — APPROVED** [review]

**Pull request merged**